manifests/server/contrib.pp

```diff
@@ -5,17 +5,17 @@
 # @param package_ensure
 #   Ensure the contrib package is installed.
 class postgresql::server::contrib (
-  String $package_name      = $postgresql::params::contrib_package_name,
-  String[1] $package_ensure = 'present'
+  Optional[String[1]] $package_name = $postgresql::params::contrib_package_name,
+  String[1] $package_ensure         = 'present'
+  Optional[Boolean] $allow_virtual  = $postgresql::params::contrib_virtual_package,
```

Comment:
I don't think this is needed anymore. If you do keep it in, please have them match. So `$postgresql::params::contrib_virtual_package` should be named `$postgresql::params::contrib_allow_virtual_package`. Another is this:
```suggestion
  Optional[Boolean] $allow_virtual  = undef,
```

If you do keep it, it also needs a puppet-strings `@param` documentation tag.

Reply:
Thanks for your help and guidance 

I tried here :) 0e5cb49ae7b3444cd5552b0a1ebcb7990ede7154